### PR TITLE
Move the mapping inputs, and add a Value mapping if values are present

### PR DESCRIPTION
**Pull request by okeefm**

### CODE REVIEW

**Comment by arscan** [comment]
Could you add this to the differential as well?  It seems weird to me to have the differential not properly represent the difference from the structure definition it is derived from, even though we technically don't look there in the es6 generator.

Just move this down to below the `dfIsNew` if block, add mapping array to `df` if it doesn't exist, and push in the same mappings to `df` as `ss`.  

Take care of that real quick, and the `def.value` being undefined issue, and I'll merge this in.

Thanks!

File: lib/export.js [hunk 1]
```diff
@@ -847,6 +831,17 @@ class FHIRExporter {
       return;
     }
 
+    // Insert mapping rules into the FHIR mapping element
+    ss.mapping = ss.mapping || [];
+    const isValue = sourceValue.effectiveIdentifier == def.value.identifier;
```
Comment (arscan) [inline lib/export.js]: `def.value` can be undefined.  Add something like this `def.value !== undefined &&`

Comment (okeefm) [reply lib/export.js]: Good catch. Updating now.

**Comment by okeefm** [comment]
Sure thing; I'll make that update.

**Comment by arscan** [comment]
Looks good.  Also, in case there is a question about it in the future, this appears to be the correct way to handle the differential for mapping when you add a new map.  I did a bit of research because I was wondering if you should only show new maps in the diff, or all maps.  The spec is unclear on this (at least I couldn't find information on this), but this is how us-core handles it, which is a strong precedent.

**Pull request merged**